### ISSUE TITLE
Allow users to overwrite display name of a given feature file

### ISSUE THREAD

**Issue by CaptainQuirk** [SOFTWARE-CLINIC/featurebook#61]
Hi,

We write our behat acceptance tests in __french__. However, all our coding artifacts are in english, which is problematic for the parsing of the spec directory.

Would it be possible to put translations inside the featurebook.json file ?

Thanks !

**Comment by danielpacak** [comment]
@CaptainQuirk could you give me an example of what is actually failing? AFAIK the gherkin parser used by the featurebook supports i18n. I'm using it, for example, with Polish language. You just need to specify the language code in the first line of your *.feature file.

```gherkin
# language: fr
Fonctionnalité: J'aime beaucoup Java
```

Even better you can send a PR to this repo https://github.com/SOFTWARE-CLINIC/featurebook-examples with some examples so I can see what exactly fails with you example specs written in French.

**Comment by CaptainQuirk** [comment]
Oh non, I'm not talking about the feature files themselves, I'm talking about the automatic __classifying__ of the feature files themselves to generate the left side menu.

I have a feature file called __summary.feature__ that gets transformed to __Summary__. Now, by providing an json, xml, yml or whatever, we could provide translations.

**Comment by danielpacak** [comment]
The data model returned by the featurebook-api, which has access to the featurebook.json looks as follows.

<img width="407" alt="screen shot 2016-12-09 at 23 54 28" src="https://cloud.githubusercontent.com/assets/1322923/21067440/7e1239ec-be6b-11e6-9f2e-ba1f57c456cb.png">

So for each node there's the `path` and `displayName` property. `displayName` is inferred from the file name itself so we can easily add the `displayNameMappings` property to `featurebook.json` and override display names easily.

```js
{
  "title": "Hello World",
  "version": "1.0",
  "authors" : [ ],
  "contributors": [ ],
  "displayNameMappings": {
    "somepath/subpath/foo.feature": "Give me nice name instead of `Foo`",
    "somepath/bar.feature": "Whatever instead of `Bar`"
  }
}
```

Is that what you're asking for @CaptainQuirk ?

**Comment by CaptainQuirk** [comment]
Hi there !

Sorry for the delay ! A `displayNameOverrides` could be nice but wouldn't the `featurebook.json` file become **unmanageable** ?

What about adding a **comment** similar to the `language` comment on top of the files themselves ? Something like:
```
# featurebook-name: ...
```

**Comment by danielpacak** [comment]
I agree that having `displayNameOverrides` in `featurebook.json` is not user friendly hence I'm going to handle a special comment in Gherkin to support such overwrites:

```gherkin
# featurebookDisplayName: Simple Feature Overwrite
Feature: Simple feature

  Scenario: Simple scenario
    Given step 1
    When step 2
    Then step 3
```

**Comment by danielpacak** [comment]
Fixed by e1ac4e70b012821d8d49844189e809190ddbd0d1
